DataFormats/TrackerRecHit2D/interface/FastTrackerRecHit.h

```diff
@@ -22,7 +22,7 @@ namespace fastTrackerRecHitType {
     siStripProjectedStereo2D = 5,
   };
   inline trackerHitRTTI::RTTI rtti(HitType hitType) {
-    if (hitType >= 0 && hitType <= 2)
+    if (hitType >= siPixel && hitType <= siStrip2D)
```

Comment:
So comparing to literal `0` is bad, but comparing to `siPixel` that has a value `0` is fine? I mean, `hitType >= siPixel` is still a tautology assuming the caller passes only valid `HitType` values.

(I'm not objecting the proposed fix, just wondering out loud)

Reply:
yes @makortel , at least llvm does not complain any more. May be the check should be change to
```
if (hitType <= siStrip2D)
or
if (hitType <= 2)
```

Reply:
I agree this behavior looks a bit weird. However, since LLVM does not complain anymore, I would keep `if (hitType >= siPixel && hitType <= siStrip2D)`. It seems more clear to me.